### PR TITLE
feat: use kaniko cache for builds

### PR DESCRIPTION
**Pull request by torfjor**

### CODE REVIEW

**Comment by torfjor** [comment]
Not quite sure how we'd leverage the cahce for `npm install --production`

Maybe copy package*.json first, and install production dependencies in a separate image, then run `npm install` and `npm run build` in a build step and copy the resulting `dist` into the production image.

File: cloudbuild.yaml [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-    args: ['build', '-t', 'gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA', '.']
-    timeout: '1500s'
-  - name: 'gcr.io/cloud-builders/docker'
-    args: ['push', 'gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA']
+  - name: 'gcr.io/kaniko-project/executor:latest'
+    args:
+      - --destination=gcr.io/$PROJECT_ID/$REPO_NAME:latest
+      - --cache=true
+      - --dockerfile=Dockerfile
+    id: build
```
Comment (mikaelbr) [inline cloudbuild.yaml:7]: No need for timeout anymore?

Comment (torfjor) [reply cloudbuild.yaml:7]: I'm not sure. One would think that a global timeout would let each step run until the global deadline is met. I'll go read the docs to make sure!

Comment (torfjor) [reply cloudbuild.yaml:7]: Looks like it does:

> Use the timeout field in a build step to set a time limit for executing the step. If you don't set this field, the step has no time limit and will be allowed to run until either it completes or the build itself times out. The timeout field in a build step must not exceed the timeout value specified for a build. timeout must be specified in seconds with up to nine fractional digits, terminated by 's'. Example: "3.5s"

**Review by mikaelbr — APPROVED** [review]

**Pull request merged**